src/sentry/utils/snuba.py

```diff
@@ -692,6 +693,7 @@ def raw_snql_query(
     # XXX (evanh): This function does none of the extra processing that the
     # other functions do here. It does not add any automatic conditions, format
     # results, nothing. Use at your own risk.
+    validate_referrer(referrer)
```

Comment:
The reason why I placed this here was execute the validation step before we write the metric to `snql.sdk.api`. Although this wouldn't matter because we are not actually rejecting queries; and when we do enforce referrers, it would be before compile time. As a follow up, I would also propose moving both the metric writes into `_apply_cache_and_build_results()` too and after the validation.

Reply:
That metric is tracking something specific, which is how often the actual SnQL function(s) are being called, vs the legacy functions that automatically convert to SnQL. Moving that metric inside the common function would complicate that, and I want the metric to be called regardless of whether the call was valid or not.